frontend/tools/translationKey/generateTranslationKeyType.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Recursive function which returns sorted array of flattened keys.
+ *
+ * For example `{a: 'abc', b: {c: 'def'}}` is returned as `['<prefix>.a', '<prefix>.b.c']`
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const flattenKeys = (obj: any, prefix: string): Array<string> => {
+  const res = Array<string>();
+  for (const key in obj) {
+    if (typeof obj[key] !== 'object') {
+      res.push(`${prefix}.${key}`);
+    } else {
+      res.push(...flattenKeys(obj[key], `${prefix}.${key}`));
+    }
+  }
+  return res;
+};
+
+/**
+ * Reads contents of translation file and returns array of flattened keys.
+ */
+const getFlattenedTranslationKeys = (locale: string, filename: string): Array<string> => {
+  const filePath = path.join(dirPath, locale, filename);
+  const fileContentString = fs.readFileSync(filePath).toString();
+  const fileContentJSON = JSON.parse(fileContentString);
+  return flattenKeys(fileContentJSON, filename.replace('.json', ''));
+};
+
+// Path to translation files from the frontend directory
+const dirPath = path.join('src', 'lib', 'i18n', 'translations');
+
+const locales = fs.readdirSync(dirPath).filter((name) => {
+  return fs.lstatSync(path.join(dirPath, name)).isDirectory();
+});
+
+// Names of translation files in the directory of the first locale
+const filenames = fs.readdirSync(path.join(dirPath, locales[0]));
+
+// Array of flattened translation keys
+const translationFilesKeys = filenames.flatMap((filename) => {
+  return getFlattenedTranslationKeys(locales[0], filename);
+});
+
+const langKeys = locales.map((locale) => `lang.${locale}`);
+const translationKeys = [...translationFilesKeys, ...langKeys].sort();
+
+const output = `export type TranslationKey =\n  | ${translationKeys.map((s) => `'${s}'`).join('\n  | ')};\n`;
+
+fs.writeFileSync(path.join('src', 'lib', 'types', 'translationKey.ts'), output);
```

Comment:
Maybe move the actual path to the top of the file as a `const`. We could also put it in a `generated` subfolder within `types` (but use an `index.ts` to export it from `types` so that the import would be `import type { TranslationKey } from '$types'`), as there might be other auto-generated types in the future as well.